src/Umbraco.Core/Services/IMediaService.cs

```diff
@@ -369,5 +370,33 @@ public interface IMediaService : IService
         /// <param name="userId">Optional id of the user creating the media item</param>
         /// <returns><see cref="IMedia"/></returns>
         IMedia CreateMediaWithIdentity(string name, int parentId, string mediaTypeAlias, int userId = 0);
+
+        /// <summary>
+        /// Gets the content of a media as a stream.
+        /// </summary>
+        /// <param name="filepath">The filesystem path to the media.</param>
+        /// <returns>The content of the media.</returns>
+        Stream GetMediaFileStream(string filepath);

```

Comment:
Guess GenerateThumbnails can be an internal thing not part of the interface, if we don't want to alter the interface _and_ we plan to stop generating tumbnails. As for the other methods... Deploy needs to have read/write access to the media files. Either it's on IMediaService... or would you rather add it to another "something"?